manager/scheduler/filter.go

```diff
@@ -217,3 +217,46 @@ func (f *ConstraintFilter) Explain(nodes int) string {
 	}
 	return fmt.Sprintf("scheduling constraints not satisfied on %d nodes", nodes)
 }
+
+// PlatformFilter selects only nodes that run the required platform.
+type PlatformFilter struct {
+	supportedPlatforms []*api.Platform
+}
+
+// SetTask returns true when the filter is enabled for a given task.
+func (f *PlatformFilter) SetTask(t *api.Task) bool {
+	placement := t.Spec.Placement
+	if placement != nil {
+		if len(placement.Platforms) > 0 {
+			// copy the platform information
+			f.supportedPlatforms = placement.Platforms
```

Comment:
Let's move this line outside above the `if len(placement.Platforms) > 0`. It shouldn't make a difference, because the filter should be disabled in this case, but it seems like a good idea to clear data that may be left over from a previous use of the filter.